ydb/core/change_exchange/change_exchange.h

```diff
@@ -8,8 +8,34 @@
 
 #include <util/generic/vector.h>
 
+namespace NKikimr {
+
+namespace NDataShard {
+
+class TChangeRecord;
+
+}
+
+namespace NReplication::NService {
+
+class TChangeRecord;
+
+}
+
+struct TBaseChangeRecordContainer {
+    virtual ~TBaseChangeRecordContainer() = default;
+    virtual TString Out() = 0;
+};
+
+template <class T>
+struct TChangeRecordContainer {};
+
+}
+
 namespace NKikimr::NChangeExchange {
 
+using TChangeRecordVector = std::variant<std::shared_ptr<TChangeRecordContainer<NDataShard::TChangeRecord>>, std::shared_ptr<TChangeRecordContainer<NReplication::NService::TChangeRecord>>>;
```

Comment:
Чтобы не слишком длинно и в будущем был понятный дифф:
```suggestion
using TChangeRecordVector = std::variant<
    std::shared_ptr<TChangeRecordContainer<NDataShard::TChangeRecord>>,
    std::shared_ptr<TChangeRecordContainer<NReplication::NService::TChangeRecord>>
>;
```